src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCSVFileRecord.java

```diff
@@ -48,6 +48,11 @@
      * Delimiter to parse lines with.
      */
     private final String delimiter;
+    
+    /*
+     * Regex to match delimiters followed by 0 or even number of quotes
+     */
+    private final String delimiterPattern = "(?=([^\"]*\"[^\"]*\")*[^\"]*$)";
```

Comment:
Fix looks good, but either make this variable public or move it to the util class so that BulkCopyCSVTest.java doesn't need to hardcode repeat this regex again.